lib/aws/rails/sqs_active_job/executor.rb

```diff
@@ -12,18 +12,19 @@ class Executor
           max_threads: Concurrent.processor_count,
           auto_terminate: true,
           idletime: 60, # 1 minute
-          fallback_policy: :caller_runs # slow down the producer thread
-          # TODO: Consider catching the exception and sleeping instead of using :caller_runs
+          fallback_policy: :abort # Concurrent::RejectedExecutionError must be handled
         }.freeze
 
         def initialize(options = {})
           @executor = Concurrent::ThreadPoolExecutor.new(DEFAULTS.merge(options))
           @retry_standard_errors = options[:retry_standard_errors]
           @logger = options[:logger] || ActiveSupport::Logger.new($stdout)
+          @task_complete = Concurrent::Event.new
         end
 
         def execute(message)
```

Comment:
I think you can use implicit `begin` here but it will require another method. You possibly should refactor this out.